src/cargo/ops/cargo_install.rs

```diff
@@ -209,8 +209,9 @@ impl<'cfg, 'a> InstallablePackage<'cfg, 'a> {
             bail!(
                 "there is nothing to install in `{}`, because it has no binaries\n\
                  `cargo install` is only for installing programs, and can't be used with libraries.\n\
-                 To use a library crate, add it as a dependency in a Cargo project instead.",
-                pkg
+                 To use a library crate, add it as a dependency to a Cargo project with `cargo add {}`.",
+                pkg,
+                pkg.name()
```

Comment:
This could be misleading. For example, if someone runs 

```
cargo install --git https://github.com/hyperium/hyper.git
```

Cargo won't suggest `cargo add hyper` with the `--git` option. 

Moreover, should Cargo be more precise on `--version` and `--features` flag if they were provided as CLI arguments? Otherwise people blindly copy `cargo add <pkg>` might install a completely different version they desire.

Reply:
I think it is a good idea to also suggest the `--git` option. Other source options that could be relevant are `--path`, `--branch`, `--tag` and `--registry`.

For the `--version` I am not sure whether we should. My use case is to suggest `cargo add` for users that used `install` instead of `add`, but `--version` is not a `cargo add` option.

Reply:
Versions can be specified by the way around: `cargo add <name>@<version-req>`. I was saying that if someone run either `cargo install mycrate --version 1.0.0` or `cargo install mycrate@1.0.0`, we may want cargo to suggest an equivalent `cargo add mycrate@1.0.0` or else.